### DISCUSSION TITLE
Setting Partitioned Cookies

### DISCUSSION THREAD

**Discussion by bstuddard** [fastapi/fastapi#11285] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import Request, Response

@app.post("/example_endpoint/")
async def example_endpoint(request: Request, response: Response):
    # refresh_expire_datetime = generate_refresh_datetime()
    response.set_cookie(key="refresh_token", value="example_generated_refresh_token", httponly=True, secure=True, samesite='none', expires=refresh_expire_datetime.replace(tzinfo=timezone.utc))

    return {
        'example_payload': 'example_payload'
    }
```

### Description

Is there a way to set [Chips](https://developers.google.com/privacy-sandbox/3pcd/chips) partitioned cookies? Use case is a separate backend and frontend, hosted in different environments. With Google's rollout of blocking cross site content and third party cookies [see here](https://developers.google.com/privacy-sandbox/3pcd#report-issues), looking for a way to set a partitioned flag. 

### Operating System

Windows

### FastAPI Version

0.93.0

### Pydantic Version

1.10.5

### Python Version

3.10.9

**Comment by tywenk** [comment]
+1 to this.

It does look like the onus is on `SimpleCookie` [implementing](https://github.com/python/cpython/pull/112714) it, before it downstream is adopted into python -> starlette -> fastapi. Ref: https://github.com/encode/starlette/pull/2501#issuecomment-1946194391

I will note that this is also becoming a requirement in our application. Our use case is embedding our application hosted at `a.com` in a client site hosted at `b.com` via an iframe. We need to securely authenticate a user from `a.com` when the top-level site is `b.com`. The only real solution, moving forward, is using CHIPS.

Google Chrome is deprecating third-party cookies (i.e. cookies with a `samesite=None` attribute) come Q1 2025. Developers will be required to migrate their implementation of third party cookies. The proposed alternative is via CHIPS, using the Partition attribute.

Ideally, the API could look like:
```
response.set_cookie(partition=True, httponly=True, secure=True, samesite='none')
```

And produce a cookie with content:
```
Set-Cookie: SameSite=None; Secure; HttpOnly; Path=/; Partitioned;
```

**Comment by aaronted009** [comment]
This moved forward on starlette end;  see https://github.com/encode/starlette/pull/2501. Only supported in Python3.14+.
